src/app/projects/edit-project/edit-project.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+
+  beforeEach(async(() => {
+    spyOn(projectsServiceStub, 'updateProject')
+      .and.returnValue(of<Project>(
+      {
+        id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17759',
+        name: 'Changed name',
+        customer: 'Changed customer',
+        industrySector: 'Changed industrySector',
+        description: 'Changed description',
+        creationDate: new Date(),
+        lastModifiedDate: new Date()
+      }));
+    TestBed.configureTestingModule({
+      imports: [
+        BrowserAnimationsModule,
+        LayoutModule,
+        FlexLayoutModule,
+        ReactiveFormsModule,
+        AppMaterialModule
+      ],
+      declarations: [ EditProjectComponent ],
+      providers: [
+        GlobalErrorHandlerService,
+        { provide: ProjectsService, useValue: projectsServiceStub },
+        { provide: MatBottomSheetRef, useValue: bottomSheetStub },
+        { provide: MAT_BOTTOM_SHEET_DATA, useValue: projectTestData }
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(EditProjectComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should initialize the edit-project form', () => {
+    expect(component.projectForm.get('name').value).toBe(projectTestData.name);
+    expect(component.projectForm.get('customer').value).toBe(projectTestData.customer);
+    expect(component.projectForm.get('industrySector').value).toBe(projectTestData.industrySector);
+    expect(component.projectForm.get('description').value).toBe(projectTestData.description);
+  });
+
+  it('should update the edit-project form', async(() => {
+    component.projectForm.get('name').setValue('Changed name');
+    component.projectForm.get('customer').setValue('Changed customer');
+    component.projectForm.get('industrySector').setValue('Changed industrySector');
+    component.projectForm.get('description').setValue('Changed description');
+    component.editProject();
+
+    fixture.whenStable().then(() => {
+      fixture.detectChanges();
+      expect(component.projectForm.get('name').value).toBe('Changed name');
+      expect(component.projectForm.get('customer').value).toBe('Changed customer');
+      expect(component.projectForm.get('industrySector').value).toBe('Changed industrySector');
+      expect(component.projectForm.get('description').value).toBe('Changed description');
+
+      const expectedRequestData: Project = {
+        id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17759',
+        name: 'Changed name',
+        customer: 'Changed customer',
+        industrySector: 'Changed industrySector',
+        description: 'Changed description',
+      } as Project;
+
+      expect(projectsServiceStub.updateProject).toHaveBeenCalledWith(expectedRequestData);
+    });
+  }));
+
+  it('should not call the createProject method when name is empty', async(() => {
```

Comment:
I think this test is quite a synthetic one. It is the component template who is responsible to prevent us from updating of a project when the name is empty. If we want to make sure we cannot perform an update of a project when the name is empty, we need to simulate a click on `update project` button within the template and check if `component.editProject()` was called.

Reply:
done